src/test/java/edu/harvard/iq/dataverse/engine/command/impl/CreateNewDataFilesTest.java

```diff
@@ -0,0 +1,261 @@
+
+            assertThat(result.getErrors()).hasSize(0);
+            assertThat(result.getDataFiles().stream().map(dataFile ->
+                dataFile.getFileMetadata().getDirectoryLabel() + "/" + dataFile.getDisplayName()
+            )).containsExactlyInAnyOrder(
+                "DD-1576/goodbye.txt", "DD-1576/hello.txt"
+            );
+            var storageIds = result.getDataFiles().stream().map(DataFile::getStorageIdentifier).toList();
+            assertThat(tempDir.toFile().list())
+                .containsExactlyInAnyOrderElementsOf(storageIds);
+        }
+    }
+
+    @Test
+    @JvmSetting(key = JvmSettings.FILES_DIRECTORY, value = "target/test/CreateNewDataFilesTest/tmp")
+    public void execute_rezips_sets_of_shape_files_from_uploaded_zip() throws Exception {
+        var tempDir = testDir.resolve("tmp/temp");
+        createDirectories(tempDir);
+
+        mockTmpLookup();
+        var cmd = createCmd("src/test/resources/own-cloud-downloads/shapes.zip", mockDatasetVersion(), 1000L, 500L);
+        var ctxt = mockCommandContext(mockSysConfig(false, 100000000L, MD5, 10));
+        try (var mockedJHoveFileType = Mockito.mockStatic(JhoveFileType.class)) {
+            mockedJHoveFileType.when(JhoveFileType::getJhoveConfigFile).thenReturn("conf/jhove/jhove.conf");
+
+            // the test
+            var result = cmd.execute(ctxt);
+
+            assertThat(result.getErrors()).hasSize(0);
+            assertThat(result.getDataFiles().stream().map(dataFile ->
+                (dataFile.getFileMetadata().getDirectoryLabel() + "/" + dataFile.getDisplayName())
+                    .replaceAll(".*temp/shp_[-0-9]*/", "")
+            )).containsExactlyInAnyOrder(
+                "dataDir/shape1.zip",
+                "dataDir/shape2/shape2",
+                "dataDir/shape2/shape2.pdf",
+                "dataDir/shape2/shape2.txt",
+                "dataDir/shape2/shape2.zip",
+                "dataDir/extra/shp_dictionary.xls",
+                "dataDir/extra/notes",
+                "dataDir/extra/README.MD"
+            );
+            var storageIds = result.getDataFiles().stream().map(DataFile::getStorageIdentifier).toList();
+            assertThat(tempDir.toFile().list())
+                .containsExactlyInAnyOrderElementsOf(storageIds);
+        }
+    }
+
+    @Disabled("Too slow. Intended for manual execution.")
+    @Test
+    @JvmSetting(key = JvmSettings.FILES_DIRECTORY, value = "/tmp/test/CreateNewDataFilesTest/tmp")
+    public void extract_zip_performance() throws Exception {
+        /*
+         Developed to test performance difference between the old implementation with ZipInputStream and the new ZipFile implementation.
+         Play with numbers depending on:
+         - the time you want to spend on this test
+         - how much system stress you want to examine
+        */
+        var nrOfZipFiles = 20;
+        var avgNrOfFilesPerZip = 300;
+        var avgFileLength = 5000;
```

Comment:
results:

    ZipFile       :: Total time: 150,909ms; nr of zips 20 total nr of files 6,404; total file size 32,030,650
    ZipInputStream:: Total time: 148,432ms; nr of zips 20 total nr of files 6,211; total file size 31,570,383